### PR TITLE
feat: Distinguish non-existent repetitions from empty values in enrollment analytics api [2.41-DHIS2-14420]

### PR DESCRIPTION
**Pull request by d-bernat**
Please see https://dhis2.atlassian.net/browse/DHIS2-14420

### CODE REVIEW

**Review by gnespolino — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java [hunk 1]
```diff
@@ -203,8 +204,22 @@ private boolean addValueMetaInfo( Grid grid, SqlRowSet rowSet, String columnName
...
+                Map<String, Object> row = rowContext.get( gridRowIndex );
+
+                if ( row == null )
+                {
+                    row = new HashMap<>();
+                }
+
+                Map<String, GridValueStatus> colValueType = new HashMap<>();
+
+                colValueType.put( "type", getGridValueStatus( isDefined, isSet ) );
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java:218]: Maybe "type" is too generic...?
What about something like "eventRepertionPresence", "eventRepertionValue", etc.?

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java [hunk 2]
```diff
@@ -156,13 +162,87 @@ private void getEnrollments( EventQueryParams params, Grid grid, String sql, boo
...
+            {
+                // when .exists extension of column name does not indicate boolean flag,
+                // value will not be added and method returns false
+            }
+        }
+
+        return false;
+    }
+
+    private RepeatableStageValueStatus getGridValueStatus( boolean isDefined, boolean isSet )
```
Comment (maikelarabori) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java:236]: This method could be moved to the Enum, so you could have
```
RepeatableStageValueStatus.of (boolean isDefined, boolean isSet)
```

Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java]: Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java]: There is repeatableStageValueStatus

**Review by maikelarabori — APPROVED** [review]

**Pull request merged**